### PR TITLE
Update head.py

### PR DESCRIPTION
**Pull request by printdhruv**
Switching to tf.summary.scalar

### CODE REVIEW

File: tensorflow/contrib/learn/python/learn/estimators/head.py [hunk 1]
```diff
@@ -37,7 +37,8 @@
 from tensorflow.python.framework import sparse_tensor
 from tensorflow.python.ops import array_ops
 from tensorflow.python.ops import control_flow_ops
-from tensorflow.python.ops import logging_ops
```
Comment (drpngx) [inline tensorflow/contrib/learn/python/learn/estimators/head.py]: please remove entirely if not needed.

File: tensorflow/contrib/learn/python/learn/estimators/head.py [hunk 2]
```diff
@@ -37,7 +37,8 @@
...
+#from tensorflow.python.ops import logging_ops
+from tensorflow.python import summary as logging_ops
```
Comment (drpngx) [inline tensorflow/contrib/learn/python/learn/estimators/head.py]: please use summary not logging_ops

**Review by drpngx — CHANGES_REQUESTED** [review]

File: tensorflow/contrib/learn/python/learn/estimators/head.py (hunk 1 above)
Comment (printdhruv) [inline tensorflow/contrib/learn/python/learn/estimators/head.py]: Hi,
       Thanks for correction. Please review it.

**Comment by drpngx** [comment]
Jenkins, test this please.

**Review by drpngx — APPROVED** [review]

**Comment by drpngx** [comment]
```
==================== Test output for //bazel_pip/tensorflow/contrib/learn:head_test:
2017-07-01 17:37:31.975956: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
2017-07-01 17:37:31.976063: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2017-07-01 17:37:31.976098: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
2017-07-01 17:37:31.976113: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
2017-07-01 17:37:31.976119: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
WARNING:tensorflow:Casting <dtype: 'float32'> labels to bool.
WARNING:tensorflow:Casting <dtype: 'float32'> labels to bool.
/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu-python3/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-py3-opt/bin/bazel_pip/tensorflow/contrib/learn/head_test.runfiles/org_tensorflow/bazel_pip/tensorflow/contrib/learn/python/learn/estimators/head_test.py:717: DeprecationWarning: Please use assertEqual instead.
  k: v[0] for k, v in six.iteritems(model_fn_ops.output_alternatives)
F..--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.4/logging/__init__.py", line 978, in emit
    msg = self.format(record)
  File "/usr/lib/python3.4/logging/__init__.py", line 828, in format
    return fmt.format(record)
  File "/usr/lib/python3.4/logging/__init__.py", line 565, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.4/logging/__init__.py", line 328, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
```

File: tensorflow/contrib/learn/python/learn/estimators/head.py (hunk 1 above)
Comment (printdhruv) [inline tensorflow/contrib/learn/python/learn/estimators/head.py]: I tried to use assertEqual but 46 tests are failing. head_test.py is outdated. Would you like to give some suggestions?

**Comment by printdhruv** [comment]
@drpngx  I compiled and ran successfully head_test.py with a minor fix in `JetBrains PyCharm 2017.1.4` with `Python 3.6` and showed status `All 83 tests passed`. I am wondering how can I verify with Jenkins from my end so that I don't waste resources here.Please share me possible ways to verify changes check for all OS.

**Comment by drpngx** [comment]
Jenkins, test this please.

**Comment by drpngx** [comment]
@printdhruv thank you for being thoughtful! Jenkins tests different platforms, and it's good to check on your end as you do. For other platforms, if you want to go the extra mile, you can try docker.  The docker platforms are [checked in](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/tools/docker). But it's fine for us to test the build as long as you tested your main platform.

**Comment by alanyee** [comment]
On line 1483, you are still using logging_ops.scalar_summary.

**Comment by frankchn** [comment]
Jenkins, test this please.

**Comment by frankchn** [comment]
`//tensorflow/contrib/learn:head_test` is failing. Sample logs:

```FAIL: testMultiClassWithLabelKeysEvalAccuracy1 (__main__.MultiClassHeadTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/head_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/head_test.py", line 1444, in testMultiClassWithLabelKeysEvalAccuracy1
    _assert_summary_tags(self, ["loss"])
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/head_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/head_test.py", line 90, in _assert_summary_tags
    test_case.assertItemsEqual(expected_tags or [], actual_tags)
AssertionError: Element counts were not equal:
First has 1, Second has 0:  'loss'
First has 0, Second has 1:  u'multi_class_head/loss'```

**Comment by printdhruv** [comment]
@frankchn @alanyee @drpngx 
I replaced `logging_ops.scalar(
          _summary_key(head_name, mkey.LOSS), weighted_average_loss)		          _summary_key(head_name, mkey.LOSS), weighted_average_loss)`
 
with the following statement.

 `tensorflow.summary.scalar(
        name=_summary_key(head_name, mkey.LOSS), tensor=weighted_average_loss,collections=None)`

What I think the` AssertionError: Element counts were not equal:` is generated is due to the eval and loss is in the different graph according to author's comment on line `623-624`.
```

# Uses the deprecated API to set the tag explicitly.
# Without it, training and eval losses will show up in different graphs.

```
If I am wrong then please give your suggestions.

**Comment by frankchn** [comment]
@printdhruv, can you make @alanyee's change first and rebase to master? `logging_ops` is still used on line 1483.

**Review by printdhruv — COMMENTED** [review]
Updated

**Comment by frankchn** [comment]
Jenkins, test this please.

**Comment by printdhruv** [comment]
Resolved [https://github.com/tensorflow/tensorflow/pull/11526](https://github.com/tensorflow/tensorflow/pull/11526)

**Comment by alanyee** [comment]
@printdhruv I haven't fixed the deprecated tf.summary yet.

**Pull request closed without merging**